Broken test: LargestFirstMemoryManagerTest

**Test name(s)**
 - org.apache.accumulo.tserver.LargestFirstMemoryManagerTest.test

**Describe the failure observed**
```java
java.lang.AssertionError: expected:<1;f<> but was:<1;e<>
	at org.apache.accumulo.tserver.LargestFirstMemoryManagerTest.test(LargestFirstMemoryManagerTest.java:117)
```

**Testing Environment:**
 - Version of this project: 2.1.0-SNAPSHOT
 - First commit known to fail (or current commit): 5cb30ed0e9a7f3d481ea49dd126d64cc7ab9ef53
 - Jenkins shows:
```
Executing Maven:  -B -f /home/jenkins/workspace/Accumulo/main/pom.xml -V -e -ntp clean verify -Dstyle.color=always -Dfailsafe.rerunFailingTestsCount=5 -Dsurefire.forkCount=2C
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/asfpackages/maven/apache-maven-3.6.3
Java version: 11.0.6, vendor: Oracle Corporation, runtime: /usr/local/asfpackages/java/jdk-11.0.6
Default locale: en_US, platform encoding: ISO-8859-1
OS name: "linux", version: "4.15.0-74-generic", arch: "amd64", family: "unix"
```

I saw this happen during a failure:

<pre>
[INFO] Running org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest
[INFO] Running org.apache.accumulo.tserver.TabletResourceManagerTest
[ERROR] WARNING: An illegal reflective access operation has occurred
[ERROR] WARNING: Illegal reflective access by org.easymock.cglib.core.ReflectUtils$1 (file:/home/runner/.m2/repository/org/easymock/easymock/4.2/easymock-4.2.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
[ERROR] WARNING: Please consider reporting this to the maintainers of org.easymock.cglib.core.ReflectUtils$1
[ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[ERROR] WARNING: All illegal access operations will be denied in a future release
[ERROR] WARNING: An illegal reflective access operation has occurred
[ERROR] WARNING: Illegal reflective access by org.easymock.cglib.core.ReflectUtils$1 (file:/home/runner/.m2/repository/org/easymock/easymock/4.2/easymock-4.2.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
[ERROR] WARNING: Please consider reporting this to the maintainers of org.easymock.cglib.core.ReflectUtils$1
[ERROR] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[ERROR] WARNING: All illegal access operations will be denied in a future release
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.674 s - in org.apache.accumulo.tserver.TabletResourceManagerTest
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.007 s <<< FAILURE! - in org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest
[ERROR] org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest.test  Time elapsed: 1.721 s  <<< FAILURE!
java.lang.AssertionError: expected:<1;f<> but was:<1;e<>
	at org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest.test(LargestFirstMemoryManagerTest.java:118)
</pre>

I don't know what is going on but it appears that multiple tests are running at the same time and the other test completes before the LargestFirstMemoryManagerTest.

After some experimenting with the ZERO_TIME value of the test and printing different values coming out of the `LargestFirstMemoryManager`, I think I have figured out how the test occasionally fails. As to exactly how all the numbers at play come to cause it to fail, is another story. Here are the strange circumstances I created to lead to the failure.

I modified the ZERO_TIME of the test to be _exactly_ 4 minutes less than the `LargestFirstMemoryManager`:
<pre>
public class LargestFirstMemoryManagerTest {

  private static final long ZERO = LargestFirstMemoryManager.ZERO_TIME - (4 * 60 * 1000);
</pre>

This will cause the test to always fail on my machine at the same spot we have seen above and with the exact same AssertionError message:
<pre>
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.008 s <<< FAILURE! - in org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest
[ERROR] org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest.test  Time elapsed: 0.93 s  <<< FAILURE!
java.lang.AssertionError: expected:<1;f<> but was:<1;e<>
	at org.apache.accumulo.tserver.memory.LargestFirstMemoryManagerTest.test(LargestFirstMemoryManagerTest.java:118)
</pre>

The weird timing difference combined with the following oddities will lead to this failure:
- The math used in the `LargestFirstMemoryManager` to calculate timeMemoryLoad, which is used as a key in the `LargestMap` largestMemTablets
- The fact that the `LargestMap` inner class uses an `ArrayList` to store multiple tablets keyed at the the same Long
- The final loop in the `LargestFirstMemoryManager` will end as soon as it hits the desired number of tablets

This can be seen in these log statements:
<pre>
2021-02-24T11:15:57,223 [memory.LargestFirstMemoryManager] INFO : largestMemTablets.put(timeMemoryLoad 446267447, 1;a<)
2021-02-24T11:15:57,224 [memory.LargestFirstMemoryManager] INFO : largestMemTablets.put(timeMemoryLoad 892534895, 1;b<)
2021-02-24T11:15:57,224 [memory.LargestFirstMemoryManager] INFO : largestMemTablets.put(timeMemoryLoad 892534896, 1;c<)
2021-02-24T11:15:57,224 [memory.LargestFirstMemoryManager] INFO : largestMemTablets.put(timeMemoryLoad 892534897, 1;d<)
2021-02-24T11:15:57,225 [memory.LargestFirstMemoryManager] INFO : <b>largestMemTablets.put(timeMemoryLoad 892534898, 1;e<)</b>
2021-02-24T11:15:57,225 [memory.LargestFirstMemoryManager] INFO : <b>largestMemTablets.put(timeMemoryLoad 892534898, 1;f<)</b>
2021-02-24T11:15:57,225 [memory.LargestFirstMemoryManager] INFO : <b>adding 1;f< to list for 892534898</b>
2021-02-24T11:15:57,226 [memory.LargestFirstMemoryManager] INFO : largestMemTablets.put(timeMemoryLoad 892534899, 1;g<)
2021-02-24T11:15:57,226 [memory.LargestFirstMemoryManager] INFO : Last entry 892534899 has list of size = 1
2021-02-24T11:15:57,226 [memory.LargestFirstMemoryManager] INFO : COMPACTING 1;g<  total = 6,979,321,877 ingestMemory = 6,979,321,877
2021-02-24T11:15:57,227 [memory.LargestFirstMemoryManager] INFO : chosenMem = 1,073,741,830 chosenIT = -240.00 load 892,534,899
2021-02-24T11:15:57,227 [memory.LargestFirstMemoryManager] INFO : <b>Last entry 892534898 has list of size = 2</b>
2021-02-24T11:15:57,227 [memory.LargestFirstMemoryManager] INFO : COMPACTING 1;e<  total = 6,979,321,877 ingestMemory = 6,979,321,877
2021-02-24T11:15:57,228 [memory.LargestFirstMemoryManager] INFO : chosenMem = 1,073,741,828 chosenIT = -240.00 load 892,534,898
2021-02-24T11:15:57,228 [memory.LargestFirstMemoryManager] INFO : returning tabletsToMinorCompact with 2 tablets
</pre>

During a failure, the final loop in `LargestFirstMemoryManager` will take the next tablet in the list of the lastEntry, which may or may not be the next largest tablet since the backing `ArrayList` of `LargestMap` is not sorted. During a success, each entry has an `ArrayList` of size one, stored at distinct Long keys.

--- a/server/tserver/src/main/java/org/apache/accumulo/tserver/memory/LargestFirstMemoryManager.java
+++ b/server/tserver/src/main/java/org/apache/accumulo/tserver/memory/LargestFirstMemoryManager.java
@@ -44,7 +44,7 @@ import org.slf4j.LoggerFactory;
 public class LargestFirstMemoryManager {
 
   private static final Logger log = LoggerFactory.getLogger(LargestFirstMemoryManager.class);
-  private static final long ZERO_TIME = System.currentTimeMillis();
+  static final long ZERO_TIME = System.currentTimeMillis();
   private static final int TSERV_MINC_MAXCONCURRENT_NUMWAITING_MULTIPLIER = 2;
   private static final double MAX_FLUSH_AT_ONCE_PERCENT = 0.20;
 

--- a/server/tserver/src/test/java/org/apache/accumulo/tserver/memory/LargestFirstMemoryManagerTest.java
+++ b/server/tserver/src/test/java/org/apache/accumulo/tserver/memory/LargestFirstMemoryManagerTest.java
@@ -42,9 +42,10 @@ public class LargestFirstMemoryManagerTest {
   @Rule
   public Timeout timeout = Timeout.seconds(60);
 
-  private static final long ZERO = System.currentTimeMillis();
+  private static final long ZERO = LargestFirstMemoryManager.ZERO_TIME;
   private static final long LATER = ZERO + 20 * 60 * 1000;
   private static final long ONE_GIG = 1024 * 1024 * 1024;
+  private static final long ONE_MEG = 1024 * 1024;
   private static final long HALF_GIG = ONE_GIG / 2;
   private static final long QGIG = ONE_GIG / 4;
   private static final long ONE_MINUTE = 60 * 1000;
@@ -90,45 +91,47 @@ public class LargestFirstMemoryManagerTest {
     mgr = new LargestFirstMemoryManagerUnderTest();
     mgr.init(context);
     tabletsToMinorCompact = mgr.tabletsToMinorCompact(tablets(t(k("a"), ZERO, HALF_GIG, 0),
-        t(k("b"), ZERO, HALF_GIG + 1, 0), t(k("c"), ZERO, HALF_GIG + 2, 0),
-        t(k("d"), ZERO, HALF_GIG + 3, 0), t(k("e"), ZERO, HALF_GIG + 4, 0),
-        t(k("f"), ZERO, HALF_GIG + 5, 0), t(k("g"), ZERO, HALF_GIG + 6, 0),
-        t(k("h"), ZERO, HALF_GIG + 7, 0), t(k("i"), ZERO, HALF_GIG + 8, 0)));
+        t(k("b"), ZERO, HALF_GIG + ONE_MEG, 0), t(k("c"), ZERO, HALF_GIG + (2 * ONE_MEG), 0),
+        t(k("d"), ZERO, HALF_GIG + (3 * ONE_MEG), 0), t(k("e"), ZERO, HALF_GIG + (4 * ONE_MEG), 0),
+        t(k("f"), ZERO, HALF_GIG + (5 * ONE_MEG), 0), t(k("g"), ZERO, HALF_GIG + (6 * ONE_MEG), 0),
+        t(k("h"), ZERO, HALF_GIG + (7 * ONE_MEG), 0),
+        t(k("i"), ZERO, HALF_GIG + (8 * ONE_MEG), 0)));
     assertEquals(2, tabletsToMinorCompact.size());
     assertEquals(k("i"), tabletsToMinorCompact.get(0));
     assertEquals(k("h"), tabletsToMinorCompact.get(1));
     // one finished, one in progress, one filled up
     mgr = new LargestFirstMemoryManagerUnderTest();
     mgr.init(context);
-    tabletsToMinorCompact = mgr.tabletsToMinorCompact(
-        tablets(t(k("a"), ZERO, HALF_GIG, 0), t(k("b"), ZERO, HALF_GIG + 1, 0),
-            t(k("c"), ZERO, HALF_GIG + 2, 0), t(k("d"), ZERO, HALF_GIG + 3, 0),
-            t(k("e"), ZERO, HALF_GIG + 4, 0), t(k("f"), ZERO, HALF_GIG + 5, 0),
-            t(k("g"), ZERO, ONE_GIG, 0), t(k("h"), ZERO, 0, HALF_GIG + 7), t(k("i"), ZERO, 0, 0)));
+    tabletsToMinorCompact = mgr.tabletsToMinorCompact(tablets(t(k("a"), ZERO, HALF_GIG, 0),
+        t(k("b"), ZERO, HALF_GIG + ONE_MEG, 0), t(k("c"), ZERO, HALF_GIG + (2 * ONE_MEG), 0),
+        t(k("d"), ZERO, HALF_GIG + (3 * ONE_MEG), 0), t(k("e"), ZERO, HALF_GIG + (4 * ONE_MEG), 0),
+        t(k("f"), ZERO, HALF_GIG + (5 * ONE_MEG), 0), t(k("g"), ZERO, ONE_GIG, 0),
+        t(k("h"), ZERO, 0, HALF_GIG + (7 * ONE_MEG)), t(k("i"), ZERO, 0, 0)));
     assertEquals(1, tabletsToMinorCompact.size());
     assertEquals(k("g"), tabletsToMinorCompact.get(0));
     // memory is very full, lots of candidates
-    tabletsToMinorCompact = mgr.tabletsToMinorCompact(
-        tablets(t(k("a"), ZERO, HALF_GIG, 0), t(k("b"), ZERO, ONE_GIG + 1, 0),
-            t(k("c"), ZERO, ONE_GIG + 2, 0), t(k("d"), ZERO, ONE_GIG + 3, 0),
-            t(k("e"), ZERO, ONE_GIG + 4, 0), t(k("f"), ZERO, ONE_GIG + 5, 0),
-            t(k("g"), ZERO, ONE_GIG + 6, 0), t(k("h"), ZERO, 0, 0), t(k("i"), ZERO, 0, 0)));
+    tabletsToMinorCompact = mgr.tabletsToMinorCompact(tablets(t(k("a"), ZERO, HALF_GIG, 0),
+        t(k("b"), ZERO, ONE_GIG + ONE_MEG, 0), t(k("c"), ZERO, ONE_GIG + (2 * ONE_MEG), 0),
+        t(k("d"), ZERO, ONE_GIG + (3 * ONE_MEG), 0), t(k("e"), ZERO, ONE_GIG + (4 * ONE_MEG), 0),
+        t(k("f"), ZERO, ONE_GIG + (5 * ONE_MEG), 0), t(k("g"), ZERO, ONE_GIG + (6 * ONE_MEG), 0),
+        t(k("h"), ZERO, 0, 0), t(k("i"), ZERO, 0, 0)));
     assertEquals(2, tabletsToMinorCompact.size());
     assertEquals(k("g"), tabletsToMinorCompact.get(0));
     assertEquals(k("f"), tabletsToMinorCompact.get(1));
     // only have two compactors, still busy
-    tabletsToMinorCompact = mgr.tabletsToMinorCompact(
-        tablets(t(k("a"), ZERO, HALF_GIG, 0), t(k("b"), ZERO, ONE_GIG + 1, 0),
-            t(k("c"), ZERO, ONE_GIG + 2, 0), t(k("d"), ZERO, ONE_GIG + 3, 0),
-            t(k("e"), ZERO, ONE_GIG + 4, 0), t(k("f"), ZERO, ONE_GIG, ONE_GIG + 5),
-            t(k("g"), ZERO, ONE_GIG, ONE_GIG + 6), t(k("h"), ZERO, 0, 0), t(k("i"), ZERO, 0, 0)));
+    tabletsToMinorCompact = mgr.tabletsToMinorCompact(tablets(t(k("a"), ZERO, HALF_GIG, 0),
+        t(k("b"), ZERO, ONE_GIG + ONE_MEG, 0), t(k("c"), ZERO, ONE_GIG + (2 * ONE_MEG), 0),
+        t(k("d"), ZERO, ONE_GIG + (3 * ONE_MEG), 0), t(k("e"), ZERO, ONE_GIG + (4 * ONE_MEG), 0),
+        t(k("f"), ZERO, ONE_GIG, ONE_GIG + (5 * ONE_MEG)),
+        t(k("g"), ZERO, ONE_GIG, ONE_GIG + (6 * ONE_MEG)), t(k("h"), ZERO, 0, 0),
+        t(k("i"), ZERO, 0, 0)));
     assertEquals(0, tabletsToMinorCompact.size());
     // finished one
-    tabletsToMinorCompact = mgr.tabletsToMinorCompact(
-        tablets(t(k("a"), ZERO, HALF_GIG, 0), t(k("b"), ZERO, ONE_GIG + 1, 0),
-            t(k("c"), ZERO, ONE_GIG + 2, 0), t(k("d"), ZERO, ONE_GIG + 3, 0),
-            t(k("e"), ZERO, ONE_GIG + 4, 0), t(k("f"), ZERO, ONE_GIG, ONE_GIG + 5),
-            t(k("g"), ZERO, ONE_GIG, 0), t(k("h"), ZERO, 0, 0), t(k("i"), ZERO, 0, 0)));
+    tabletsToMinorCompact = mgr.tabletsToMinorCompact(tablets(t(k("a"), ZERO, HALF_GIG, 0),
+        t(k("b"), ZERO, ONE_GIG + ONE_MEG, 0), t(k("c"), ZERO, ONE_GIG + (2 * ONE_MEG), 0),
+        t(k("d"), ZERO, ONE_GIG + (3 * ONE_MEG), 0), t(k("e"), ZERO, ONE_GIG + (4 * ONE_MEG), 0),
+        t(k("f"), ZERO, ONE_GIG, ONE_GIG + (5 * ONE_MEG)), t(k("g"), ZERO, ONE_GIG, 0),
+        t(k("h"), ZERO, 0, 0), t(k("i"), ZERO, 0, 0)));
     assertEquals(1, tabletsToMinorCompact.size());
     assertEquals(k("e"), tabletsToMinorCompact.get(0));
 
@@ -136,7 +139,7 @@ public class LargestFirstMemoryManagerTest {
     mgr = new LargestFirstMemoryManagerUnderTest();
     mgr.init(context);
     tabletsToMinorCompact = mgr.tabletsToMinorCompact(tablets(t(k("a"), ZERO, HALF_GIG, 0),
-        t(k("b"), ZERO, HALF_GIG + 1, 0), t(k("c"), ZERO, HALF_GIG + 2, 0),
+        t(k("b"), ZERO, HALF_GIG + ONE_MEG, 0), t(k("c"), ZERO, HALF_GIG + (2 * ONE_MEG), 0),
         t(k("d"), ZERO, 0, HALF_GIG), t(k("e"), ZERO, 0, HALF_GIG), t(k("f"), ZERO, 0, HALF_GIG),
         t(k("g"), ZERO, 0, HALF_GIG), t(k("i"), ZERO, 0, HALF_GIG), t(k("j"), ZERO, 0, HALF_GIG),
         t(k("k"), ZERO, 0, HALF_GIG), t(k("l"), ZERO, 0, HALF_GIG), t(k("m"), ZERO, 0, HALF_GIG)));

